test-suite/src/test/scala/org/scalajs/testsuite/javalib/MapTest.scala

```diff
@@ -0,0 +1,604 @@
+        expect(mp.get("one")).toEqual(null)
+      }
+    } else {
+      it("should not put null values") {
+        val mp = mapFactory.empty[String, String]
+        expectNullPointerException(mp.put("one", null))
+      }
+    }
+
+
+    it("should be cleared with one operation") {
+      val mp = mapFactory.empty[String, String]
+
+      mp.put("ONE", "one")
+      mp.put("TWO", "two")
+      expect(mp.size()).toEqual(2)
+      mp.clear()
+      expect(mp.size()).toEqual(0)
+    }
+
+    it("should check contained key presence") {
+      val mp = mapFactory.empty[String, String]
+
+      mp.put("ONE", "one")
+      expect(mp.containsKey("ONE")).toBeTruthy
+      expect(mp.containsKey("TWO")).toBeFalsy
+      expect(mp.containsKey(null)).toBeFalsy
+    }
+
+    it("should check contained value presence") {
+      val mp = mapFactory.empty[String, String]
+
+      mp.put("ONE", "one")
+      expect(mp.containsValue("one")).toBeTruthy
+      expect(mp.containsValue("two")).toBeFalsy
+      expect(mp.containsValue(null)).toBeFalsy
+    }
+
+    it("should give proper Collection over values") {
+      val mp = mapFactory.empty[String, String]
+
+      mp.put("ONE", "one")
+      val values = mp.values
+      expect(values.size).toEqual(1)
+      val iter = values.iterator
+      expect(iter.hasNext).toBeTruthy
+      expect(iter.next).toEqual("one")
+      expect(iter.hasNext).toBeFalsy
+    }
+
+    it("should give proper EntrySet over key values pairs") {
+      val mp = mapFactory.empty[String, String]
+
+      mp.put("ONE", "one")
+      val entrySet = mp.entrySet
+
+      expect(entrySet.size).toEqual(1)
+
+      val iter = entrySet.iterator
+      expect(iter.hasNext).toBeTruthy
+      val next = iter.next
+      expect(iter.hasNext).toBeFalsy
+      expect(next.getKey).toEqual("ONE")
+      expect(next.getValue).toEqual("one")
+    }
+
+    it("should give proper KeySet over keys") {
+      val mp = mapFactory.empty[String, String]
+
+      mp.put("ONE", "one")
+      val keySet = mp.keySet
+
+      expect(keySet.size).toEqual(1)
+
+      val iter = keySet.iterator
+      expect(iter.hasNext).toBeTruthy
+      expect(iter.next).toEqual("ONE")
+      expect(iter.hasNext).toBeFalsy
+    }
+
+    it("should put a whole map into") {
+      val mp = mapFactory.empty[String, String]
+
+      val m = mu.Map[String, String](
+        "X" -> "y")
+      mp.putAll(mutableMapAsJavaMap(m))
+      expect(mp.size).toEqual(1)
+      expect(mp.get("X")).toEqual("y")
+
+      val nullMap = mu.Map[String, String](
+        (null: String) -> "y",
+        "X" -> "y")
+
+      if (allowsNullKeys) {
+        mp.putAll(mutableMapAsJavaMap(nullMap))
+        expect(mp.get(null)).toEqual("y")
+        expect(mp.get("X")).toEqual("y")
+      } else {
+        expectNullPointerException(mp.putAll(mutableMapAsJavaMap(nullMap)))
+      }
+    }
+
+  }
+
+  case class SimpleQueryableMap[K, V](inner: mu.HashMap[K, V])
+    extends ju.AbstractMap[K, V] {
```

Comment:
This is part of the code that I factored out from the `ConcurrentHashMap` tests that @andreaTP wrote. It seams that it is just for the `SimpleQueryableMap.apply` constructor, I'll remove the `case` and use `new` to instantiate.